posts/2024-02-19-waku-network-tech-overview.mdx

```diff
@@ -0,0 +1,172 @@
+		
+One issue with sharding is that users and applications must know what shard to use.
+		
+One application could arbitrarily decide what shard they want to use, but that may take a lot of work, especially when adding new shards. It is also one more decision for the developer to make; we prefer to make the developer experience as easy as possible. It also allows developers to build applications that spread on several shards without delegating shard choices to the user.
+		
+Autosharding is a simple protocol that dispatches messages on a shard based on the application.
+### Message Routing - RLN Relay
+		
+Waku is generalised and privacy-preserving, meaning any kind of payload can be transported. Hence, there is no strict definition of "spam". Whether the message contains a [meme](https://github.com/b4s36t4/waku-meme) or a [zk note for private DeFi](https://railgun.ch/#/), Waku shouldn't even know the content; metadata and data should remain private.
+		
+Hence, there is a risk of someone pushing gigabits of data to the network. This can be a problem at different levels:
+		
+- Bandwidth usage: This may lead to a user's bandwidth being hogged and affecting other services (streaming, staking) or a surprise bill.
+- Connectivity: if a node does not have enough bandwidth to send/receive all messages on the gossipsub layer, then their behaviour may be seen as incorrect by other nodes that may disconnect from it.
+- Reliability: If the traffic is higher than the available bandwidth, then a node may not be able to send and receive messages reliably.
+- Other resources: memory usage is correlated with the traffic, and so is disk space for a store service.
+		
+So rather than defining what a "spam" message looks like, we introduced rate limit on the network to enable a fair use of Waku, with a capping of the bandwidth usage per shard.
+		
+This is done with RLN, or Rate Limiting Nullifier, which limits the rate of messages a given publisher sends.
+It is currently set to 1 msg/s. Coupled with a maximum message size (150kB) and a maximum number of publishers (80k, TBD), we can assume a maximum bandwidth usage per shard (around 10Mbps).
+		
+Rate-limiting publishers in a censorship-resistant and private manner is difficult; this is why we are using zero-knowledge technology:
+		
+1. Users push their RLN credentials to a smart contract (currently on Ethereum Sepolia testnet).
+2. Nodes keep track of all registered memberships on the contract.
+3. When sending a message, the user attaches an RLN proof to the message with the current epoch (timestamp in seconds).
+4. Nodes can verify the proof without being able to correlate the user's Ethereum address (used on the smart contract) with the message being sent, preserving anonymity.
+5. If a user attempts to send more than 1 msg/s, nodes can detect this and drop the message surplus or spam, protecting the network.
+
+### Servicing Mostly-offline and Resource-restricted Devices
```

Comment:
I believe there was some acid info guideline that we should only capitalized first word in title?